Examples/clim-fig.lisp

```diff
@@ -375,8 +375,7 @@
   (let* ((output-history (clim-fig-output-record *application-frame*))
 	 (record-count (output-record-count output-history))
          (record (and (not (zerop record-count))
-		      (elt (output-record-children output-history)
-			   (1- record-count)))))
+		      (first (output-record-children output-history)))))
```

Comment:
This fix is not correct, because it depends on assumption, that output-record-children are in some particular order. There is no guarantee for that. In fact this code is correct for standard-sequence-output-record (where add is an operation `vector-push-extend`). In case of tree-output-record (for some reason) it is reversed and that may require further investigation. It is not stated in the spec what is the order (http://bauhh.dyndns.org:8000/clim-spec/16-2.html#_859).

Also record-count in this commit is left unusued (then there would be no need to investigate).

A correct fix would be collecting output records on a separate list where the order is known and access them from there (redo command does something in this spirit).

To have access to the output-record when it is drawn you need to use with-output-to-output-record and then add it to the stream.

That's at least my impression how this should be tackled.

Reply:
Okay, 2f3fa9c fixes the behaviour without any dependence on the internals of tree-output-record, per your suggestion.

Reply:
please squash commits so there is only one.